pmdarima/model_selection/_split.py

```diff
@@ -242,7 +242,7 @@ def _iter_train_test_indices(self, y, X):
         # Determine the number of iterations that will take place. Must
         # guarantee that the forecasting horizon will not over-index the series
         all_indices = np.arange(n_samples)
-        for train_step_size in range(0, n_samples - h - initial, step):
+        for train_step_size in range(0, n_samples - h - initial + 1, step):
```

Comment:
I am not entirely convinced it's this simple. Can you add some tests like [this](https://github.com/alkaline-ml/pmdarima/blob/master/pmdarima/model_selection/tests/test_split.py#L107)?

Reply:
Thinking we may have to do something more like https://github.com/alkaline-ml/pmdarima/pull/388/files#diff-583c6f60a96024f91b533cdf7a5c041eR341

Reply:
Sure let me look into that

Reply:
Added in 61983da. Basically just coped what was there. You might confirm that the test I used is actually testing what we want; I have little domain knowledge here